code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/service/impl/HealthcheckServiceImpl.java

```diff
@@ -259,17 +260,33 @@ protected void createHealthCheckHostMaps(HealthcheckInstanceType instanceType, l
                 HealthcheckInstance.class, healthInstance.getId());
         List<? extends Host> availableActiveHosts = allocatorDao.getActiveHosts(healthInstance.getAccountId());
         
+        // skip non-linux hosts
+        Iterator<? extends Host> it = availableActiveHosts.iterator();
+        while (it.hasNext()) {
+            Host host = it.next();
+            Map<String, Object> data = (Map<String, Object>) host.getData();
```

Comment:
@LLParse there is a slightly better way to retrieve field from the "data
 field of the object: https://github.com/rancher/cattle/blob/ffd037c0fd465ba3f5fa15d6dad0567f346ae5e2/code/iaas/logic/src/main/java/io/cattle/platform/process/host/HostLabelReconcile.java#L46